tree/tree/src/TLeafElement.cxx

```diff
@@ -52,6 +53,47 @@ TLeafElement::TLeafElement(TBranch *parent, const char *name, Int_t id, Int_t ty
       if ((bareType >= TVirtualStreamerInfo::kUChar && bareType <= TVirtualStreamerInfo::kULong)
           || bareType == TVirtualStreamerInfo::kULong64)
       SetUnsigned();
+
+      auto bareTypeCopy = static_cast<EDataType>(bareType);
+      switch (bareTypeCopy) {
+          case kChar_t: // fall-through
+          case kUChar_t: // fall-through
+          case kchar: // fall-through
+          case kBool_t:
+              fLenType = 1;
+              break;
+          case kShort_t: // fall-through
+          case kUShort_t: // fall-through
+          case kFloat16_t:
+              fLenType = 2;
+              break;
+          case kFloat_t: // fall-through
+          case kDouble32_t: // fall-through
+          case kInt_t: // fall-through
+          case kUInt_t:
+              fLenType = 4;
+              break;
+          case kLong_t: // fall-through
+          case kULong_t: // fall-through
```

Comment:
@pcanal - can you provide advice on the actual sizing of the data members?

For types like `kLong_t` where the size differs between the platforms, what is serialized into the buffer?  Seems we should have a platform-invariant storage size, no?

Would seem strange if a buffer of `kLong_t` written on Windows is unreadable on Unix because `sizeof(long)` differs between the two platforms.